### ISSUE TITLE
Processor enters endless recursion (2.2-beta-3)

### ISSUE THREAD

**Issue by cal101** [immutables/immutables#324]
Hi,

I am playing for the first time with immutable library and it's looks really great.
Thanks for sharing!
I am trying to make some object tree deep immutable. 
Some nodes reference other nodes of same type.
The following is a simplified case I observed.

The code
```java
@Value.Immutable
@Value.Style(deepImmutablesDetection = true)
public interface LinkNode {
    String getName();
    LinkNode getNext();
}
```

leads to
```
...
    [javac] 	at org.immutables.value.internal.$processor$.meta.$ValueAttribute.initAndValidate($ValueAttribute.java:906)
    [javac] 	at org.immutables.value.internal.$processor$.meta.$AccessorAttributesCollector.collect($AccessorAttributesCollector.java:96)
    [javac] 	at org.immutables.value.internal.$processor$.meta.$ValueTypeComposer.compose($ValueTypeComposer.java:66)
    [javac] 	at org.immutables.value.internal.$processor$.meta.$Proto$Environment.composeValue($Proto.java:303)
    [javac] 	at org.immutables.value.internal.$processor$.meta.$ValueAttribute.initAttributeValueType($ValueAttribute.java:942)
    [javac] 	at org.immutables.value.internal.$processor$.meta.$ValueAttribute.initAndValidate($ValueAttribute.java:906)
    [javac] 	at org.immutables.value.internal.$processor$.meta.$AccessorAttributesCollector.collect($AccessorAttributesCollector.java:96)
    [javac] 	at org.immutables.value.internal.$processor$.meta.$ValueTypeComposer.compose($ValueTypeComposer.java:66)
    [javac] 	at org.immutables.value.internal.$processor$.meta.$Proto$Environment.composeValue($Proto.java:303)
    [javac] 	at org.immutables.value.internal.$processor$.meta.$ValueAttribute.initAttributeValueType($ValueAttribute.java:942)
    [javac] 	at org.immutables.value.internal.$processor$.meta.$ValueAttribute.initAndValidate($ValueAttribute.java:906)
    [javac] 	at org.immutables.value.internal.$processor$.meta.$AccessorAttributesCollector.collect($AccessorAttributesCollector.java:96)
    [javac] 	at org.immutables.value.internal.$processor$.meta.$ValueTypeComposer.compose($ValueTypeComposer.java:66)
    [javac] 	at org.immutables.value.internal.$processor$.meta.$Proto$Environment.composeValue($Proto.java:303)
    [javac] 	at org.immutables.value.internal.$processor$.meta.$ValueAttribute.initAttributeValueType($ValueAttribute.java:942)
...
```

I won't need deep recursions if I could provide a converter function/class somehow to be used when assigning value of some class to builder field or immutable object constructor.

Have fun,
   Cal

**Comment by elucash** [comment]
Thank you for spotting this! should be easy to fix

**Comment by cal101** [comment]
I built a jar from source and can acknowledge that the bug is fixed. Thanks!
It does not seem to resolve the problem I want to solve but thats a different story.

What's the correct way to generate sources + javadoc jars for "value" package? 

I am not that familiar with mvn and "mvn package" did generate some sources jars but not for the "value"-jar.

**Comment by elucash** [comment]
released in 2.2 Thank you!
